test/cpp/tensorexpr/test_reductions.cpp

```diff
@@ -583,5 +583,178 @@ void testReorderedReductionInitializer() {
   }
 }
 
+void testReduceRfactor() {
+  KernelScope kernel_scope;
+
+  const int M = 10;
+  const int N = 10;
+  VarHandle m("m", kInt);
+  VarHandle n("n", kInt);
+
+  Buffer b(BufHandle("b", {m, n}), kFloat);
+  std::vector<float> in(M * N);
+  for (int j = 0; j < M * N; ++j) {
+    in[j] = j;
+  }
+
+  std::vector<float> intermediate(10, -2.f);
+  std::vector<float> out(1, -1.f);
+
+  Tensor* c = Reduce("sum", {}, Sum(), b, {{m, "m"}, {n, "n"}});
+  LoopNest loop({c});
+  std::vector<For*> loops = loop.getLoopStmtsFor(c);
+  auto v = loops.at(1)->var();
+  loop.rfactor(loops.at(0), v);
+  loop.prepareForCodegen();
+  Stmt* s = loop.root_stmt();
+  s = IRSimplifier::simplify(s);
+
+  SimpleIREvaluator cg(s, {b, c, m, n});
+
+  cg.call({in, out, M, N});
+  ASSERT_EQ(out[0], 4950);
```

Comment:
I don't think that it evaluates is enough of a test here, this would pass if `rfactor` did nothing

Reply:
good point, let me count the reductions to verify